homeassistant/components/knx/binary_sensor.py

```diff
@@ -27,10 +28,19 @@ def __init__(self, device: XknxBinarySensor):
     def async_register_callbacks(self):
         """Register callbacks to update hass after device was changed."""
 
-        async def after_update_callback(device):
+        async def after_update_callback(device: XknxBinarySensor):
             """Call after device was updated."""
             self.async_write_ha_state()
 
+            self.hass.bus.fire(
```

Comment:
Custom events should not be fired from an entity or entity platform. Entities already fire state change events.

This should move to another module that isn't a platform or entity. See eg the deconz integration for an example.

Reply:
Thanks for the feedback. Would it also be possible to add the `count` variable from the underlying device to the `device_state_attributes` and then use that in the normal state change events in the automations?

But yea, we can also implement it as it is done for Deconz. Which version would be considered cleaner (if the first approach would work)?

Reply:
Adding counter as a state attribute is possible. Do these devices have a state or are they stateless buttons?

Reply:
They have an internal state (on/off) - and now possibly the counter value as well.

Reply:
Ok. It's ok to add the the counter as a state attribute.

Reply:
Alright. I'll do it this way then. I'll need some time for that change since I have to retest it and also fix the docs.

Thanks!

Reply:
Should be fine now :).